python/tk_multi_publish2/api/tree.py

```diff
@@ -338,15 +340,17 @@ class _PublishTreeEncoder(json.JSONEncoder):
     """
     Implements the json encoder interface for custom publish tree serialization.
     """
-    def default(self, publish_tree):
-        return publish_tree.to_dict()
-
-
-class _PublishTreeDecoder(json.JSONDecoder):
-    """
-    Implements the json decoder interface for custom publish tree
-    deserialization.
-    """
-    def decode(self, json_str):
-        tree_dict = sgtk.util.json.loads(json_str)
-        return PublishTree.from_dict(tree_dict)
+    def default(self, data):
+        if isinstance(data, sgtk.Template):
+            return {
+                "__tk_template": True,
+                "name": data.name
+            }
+        else:
+            return super(_PublishTreeEncoder).default(data)
+
+
+def _json_to_objects(data):
+    if data.get("__tk_template", False):
+        return sgtk.platform.current_engine().sgtk.templates[data["name"]]
```

Comment:
i think we should provide a nice error message here in case the template does not exist in the config.